### DISCUSSION TITLE
AMPScript

### DISCUSSION THREAD

**Discussion by hemanth228** [SalesforceCommerceCloud/sfcc-ci#544] [Team Posts]
Hii All,
I am trying to print builtin variable outside the ampscript block but it is not printing>May i know to how to achieve this.
%%[
set @var1 = "hemanth"

]%%
%%=v[@var1]=%%

Even this test is also not working for me

**Comment by anuragkumar05** [comment]
You need to update code.
%%[
set @var1 = "hemanth"

]%%
%%=V(@var1)=%%

**Comment by hemanth228** [comment]
i have changed as u mentioned but still it is not working

**Comment by hemanth228** [comment]
>
		%%[
	set @var1 = "hemanth"
	]%%
	%%=V(@var1)=%%

**Comment by anuragkumar05** [comment]
This code is working fine for me. please publish your cloud page and try again.

**Comment by hemanth228** [comment]
![MarketingCloud](https://user-images.githubusercontent.com/48552139/56947044-964e2880-6b49-11e9-8673-14c3d39688c9.PNG)

**Comment by hemanth228** [comment]
I have published in cloud page but name is not appearing.

**Comment by hemanth228** [comment]
Snippet of code:
 %%[
    //here ProductName is inbuit value
    set @ProductName1 = ProductName
    IF LENGTH(ProductName1) > 20 THEN
    set @ProductName1 = CONCAT(SUBSTRING(ProductName1,1,10),"...")
    ENDIF
    ]%%
      <td align="left" valign="top" style="font-family:Lyon, Georgia, 'Times New Roman', Times, serif; font-size:24px; line-height:26px; color:#3F2D38; font-weight:600;height:80px" class="em_color em_center">%%=V(@ProductName1)=%%</td>
    </tr>

**Comment by anuragkumar05** [comment]
Updated code
%%[
//here ProductName is inbuit value
set @ProductName1 = ProductName// This should be string or [ProductName]
IF LENGTH(@ProductName1) > 20 THEN
set @ProductName1 = CONCAT(SUBSTRING(@ProductName1,1,10),"...")
ENDIF
]%%
%%=V(@ProductName1)=%%

**Comment by hemanth228** [comment]
i have added updated code but still there is no luck bro

**Comment by hemanth228** [comment]
![Marketingcloud2](https://user-images.githubusercontent.com/48552139/56950665-a4a14200-6b53-11e9-8c2f-2705b2ba7aa4.PNG)

**Comment by dlferro** [comment]
Is it possible for you to paste the entire code block?
